user_guide_src/source/helpers/form_helper.rst

```diff
@@ -103,16 +103,21 @@ The following functions are available:
 
             <form action="http://example.com/index.php/email/send" class="email" id="myform" method="post" accept-charset="utf-8">
 
-        If :ref:`CSRF <cross-site-request-forgery>` filter is turned on ``form_open()`` will generate CSRF field at the beginning of the form. You can specify ID of this field by passing csrf_id as one of the ``$attribute`` array:
+        If :ref:`CSRF <cross-site-request-forgery>` filter is turned on ``form_open()`` will generate CSRF field at the beginning of the form. You can specify ID of this field by passing **csrf_id** as an element of the ``$attributes`` array:
 
         .. literalinclude:: form_helper/007.php
 
         will return::
 
             <form action="http://example.com/index.php/u/sign-up" method="post" accept-charset="utf-8">
-            <input type="hidden" id="my-id" name="csrf_field" value="964ede6e0ae8a680f7b8eab69136717d">
+            <input type="hidden" id="my-id" name="csrf_test_name" value="964ede6e0ae8a680f7b8eab69136717d">
 
-        .. note:: To use auto-generation of CSRF field, you need to turn CSRF filter on to the form page. In most cases it is requested using the ``GET`` method.
+        .. note:: To use auto-generation of CSRF field, you need to turn on the :ref:`CSRF filter <enable-csrf-protection>` in **app/Config/Filters.php** file.
+            In most cases the form page is requested using the GET method. Normally, CSRF protection is required
+            for POST/PUT/DELETE/PATCH requests, but even for GET requests, CSRF filters must be enabled for pages that display Forms.
+            
+            If you enable CSRF filter with [$globals] https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html#globals), it will be active for all request types.
+            But if you enable CSRF filter with public array $methods = ['POST' => ['csrf']];, the hidden CSRF field will not be added in GET requests.
```

Comment:
Please fix the RST format.

![Screenshot 2024-05-31 17 12 12](https://github.com/codeigniter4/CodeIgniter4/assets/87955/26b0dbc7-57e8-4485-8b01-38baa921f06a)

Reply:
@kenjis I hope it is fixed now. I don't know how to preview before commit.

Reply:
See https://github.com/codeigniter4/CodeIgniter4/tree/develop/user_guide_src#setup-instructions

Reply:
Still not good.
![Screenshot 2024-06-01 8 19 06](https://github.com/codeigniter4/CodeIgniter4/assets/87955/bc1744d3-5157-4f34-805a-d7eb67070710)

Reply:
Please apply this patch:
```diff
--- a/user_guide_src/source/helpers/form_helper.rst
+++ b/user_guide_src/source/helpers/form_helper.rst
@@ -112,12 +112,17 @@ The following functions are available:
             <form action="http://example.com/index.php/u/sign-up" method="post" accept-charset="utf-8">
             <input type="hidden" id="my-id" name="csrf_test_name" value="964ede6e0ae8a680f7b8eab69136717d">
 
-        .. note:: To use auto-generation of CSRF field, you need to turn on the :ref:`CSRF filter <enable-csrf-protection>` in **app/Config/Filters.php** file.
-            In most cases the form page is requested using the GET method. Normally, CSRF protection is required
-            for POST/PUT/DELETE/PATCH requests, but even for GET requests, CSRF filters must be enabled for pages that display Forms.
-            
-            If you enable CSRF filter with `$globals <https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html#globals>`, it will be active for all request types.
-            But if you enable CSRF filter with public array $methods = ['POST' => ['csrf']];, the hidden CSRF field will not be added in GET requests.
+        .. note:: To use auto-generation of CSRF field, you need to turn on the
+            :ref:`CSRF filter <enable-csrf-protection>` in **app/Config/Filters.php**
+            file. In most cases the form page is requested using the GET method.
+            Normally, CSRF protection is required for POST/PUT/DELETE/PATCH requests,
+            but even for GET requests, CSRF filters must be enabled for pages
+            that display Forms.
+
+            If you enable CSRF filter with :ref:`filters-globals`, it will be
+            active for all request types. But if you enable CSRF filter with
+            ``public array $methods = ['POST' => ['csrf']];``, the hidden CSRF
+            field will not be added in GET requests.
 
     **Adding Hidden Input Fields**
 
diff --git a/user_guide_src/source/incoming/filters.rst b/user_guide_src/source/incoming/filters.rst
index 303f80b551..4425ce03d3 100644
--- a/user_guide_src/source/incoming/filters.rst
+++ b/user_guide_src/source/incoming/filters.rst
@@ -139,6 +139,8 @@ Filters can be specified by adding their alias to either the ``before`` or ``aft
 
 .. literalinclude:: filters/013.php
 
+.. _filters-globals:
+
 $globals
 --------
 
```